File: app/src/main/java/com/wang/mytest/network/NetworkActivity.java
package com.wang.mytest.network;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

public class NetworkActivity extends AppCompatActivity {

    private NetworkLiveModel mNetworkLiveModel;

    @Override
    protected void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mNetworkLiveModel = ViewModelProviders.of(this).get(NetworkLiveModel.class);
        mNetworkLiveModel.getNetworkLiveData().observe(this, new Observer<Boolean>() {
            @Override
            public void onChanged(Boolean isConnected) {

            }
        });
    }
}

File: app/src/main/java/com/wang/mytest/network/NetworkActivity.kt
package com.wang.mytest.network

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders

class NetworkActivity : AppCompatActivity() {

    companion object {
        private const val TAG = "NetworkActivity"
    }

    private lateinit var liveModel: NetworkLiveModel

    private val networkObserver = Observer<Boolean> { isConnected ->
        Log.d(TAG, "isConnected = $isConnected")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        liveModel = ViewModelProviders.of(this).get(NetworkLiveModel::class.java)
        liveModel.getNetworkLiveData().observeForever(networkObserver)
    }

    override fun onDestroy() {
        super.onDestroy()
        liveModel.getNetworkLiveData().removeObserver(networkObserver)
    }
}